### PR TITLE
Runtime engine detection

### PR DESCRIPTION
**Pull request by BastienDurel**
Proposition of an API to let the programmer know which crypto engine is used in libssh2

### CODE REVIEW

File: src/version.c [hunk 1]
```diff
@@ -52,3 +52,19 @@ const char *libssh2_version(int req_version_num)
...
+#if defined LIBSSH2_OPENSSL
+    return libssh2_openssl;
+#elif defined LIBSSH2_LIBGCRYPT
+    return libssh2_gcrypt;
+#elif defined LIBSSH2_MBEDTLS
+    return libssh2_mbedtls;
+#elif defined LIBSSH2_WINCNG
+    return libssh2_wincng;
+#else
+    #error "Unhandled crypto engine"
```
Comment (willco007) [inline src/version.c:68]: You can build libssh2 with no crypto backend. It should return a `none` option instead of erroring here.

**Review by willco007 — CHANGES_REQUESTED** [review]
Looks good, just needs to return a none option with no backend.

**Comment by BastienDurel** [comment]
Hello, is this good now ?

**Review by willco007 — CHANGES_REQUESTED** [review]
The next version will be 1.11, changing the version strings here would be great. My only other thought is moving the `return libssh2_no_crypto` outside of the #if/elif block, but that's my personal preference more than anything. :)

**Comment by BastienDurel** [comment]
I prefer keeping `return libssh2_no_crypto` inside the `#else` to prevent unreachable code. Some compiler may complain about it.

**Comment by BastienDurel** [comment]
Hello,

any news ?

**Review by willco007 — CHANGES_REQUESTED** [review]
libssh2_crypto_engine.3 file needs to be added to the docs/CmakeLists.txt file as well as the Makefile.am.

**Comment by BastienDurel** [comment]
Done

**Pull request merged**